gdx-backend-jtransc/src/com/badlogic/gdx/graphics/Pixmap.java

```diff
@@ -560,7 +560,25 @@ public void drawPixel(int x, int y, int color) {
 
 	native private void line(int x, int y, int x2, int y2, DrawType drawType);
 
-	native private void rectangle(int x, int y, int width, int height, DrawType drawType);
+	private void rectangle(int x, int y, int width, int height, DrawType drawType) {
+		if (format == Format.RGBA8888) {
+			if (drawType == DrawType.FILL) {
+				int widthInBytes = this.width << 2;
+				byte[] byteColor = new byte[]{(byte) ((color >> 24) & 0xFF), (byte) ((color >> 16) & 0xFF), (byte) ((color >> 8) & 0xFF), (byte) (color & 0xFF)};
+
+				for (int i = 0; i < height; ++i) {
```

Comment:
I think that probably the fastest way:

Set first top-left pixel manually or with a loop of 4 elements if it is rgba.

Then you can use System.arraycopy that supports overlapping as feature just like memcopy to fill first row. This works like rle/LZ compression.
Then you can copy first row to all pending rows with arraycopy, that will work like memcpy since it wont require overlapping.
So here it should be just one java-level loop per number of rows with arraycopy per row.

Since this is an optimization, not required now, we can do it later